pkg/api/v1/types.go

```diff
@@ -289,3 +290,16 @@ func NewVMReferenceFromName(name string) *VM {
 	vm.SetGroupVersionKind(schema.GroupVersionKind{Group: GroupVersion.Group, Kind: "VM", Version: GroupVersion.Version})
 	return vm
 }
+
+type Migration struct {
+	metav1.TypeMeta         `json:",inline"`
+	ObjectMeta              v1.ObjectMeta     `json:"metadata,omitempty"`
+	MigratingVMName         string            `json:"nodeName,omitempty"`
```

Comment:
Could you move `MigrationVMName` and `DestinationNodeSelector` into `Spec`?

I would envision a structure like this:

```
spec:
     vmName: myvm
    constraints:
       nodeSelectors:
           kubevirt.io/hostname: mynode
status:
  phase: "PreparingMigration"
  conditions[]     
```